docs/performance/proposal.md

```diff
@@ -0,0 +1,69 @@
+- We can track and detect performance regressions in builds before release.
+
+## What we do today
+
+- Roslyn
+  - Can be run on CI: **No**
+  - Can be run locally with a single script: **No**
+  - [Compiler Performance](https://github.com/dotnet/roslyn/blob/main/docs/wiki/Measuring-Compiler-Performance.md)
+    - compiler team has written scenarios in the dotnet/performance repo. The directions for these call for developers to clone the dotnet/performance repo and manually run the tests
+    - dotnet/performance benchmarks for roslyn are [here]((https://github.com/dotnet/performance/tree/main/src/benchmarks/real-world/Roslyn))
+  - [Analyzer Performance](https://microsoft.sharepoint.com/teams/managedlanguages/_layouts/15/Doc.aspx?sourcedoc={79b652be-6aa1-4feb-8d23-fa9127483ce9}&action=edit&wd=target%28Productivity%2FHelpers.one%7Caf49b9ef-72a4-4dee-9cf1-460fe552857a%2FHow%20to%20use%20AnalyzerRunner%7Cf8d125f1-83d6-47eb-8bde-09070142ceee%2F%29)
+    - There is an AnalyzerRunner commandline tool checked into dotnet/roslyn that can be used to run analyzers and validate their performance. It needs to be run in a manual fashion.
+- [ASP.NET](https://github.com/aspnet/Benchmarks/blob/main/scenarios/README.md)
+    - Can be run on CI: **Yes**
+    - Can be run locally with a single script: **No**
+    - The ASP.NET team has written a took (crank) that allows them to run benchmarks on either their local machines or remote machines using a client/server model. This does not require the user to download the dotnet/performance repository manually to run scenarios from there. Users will need to manually setup/patch runtimes with their changes but can then run them against the real benchmarks from there.
+        - [Crank](https://github.com/dotnet/crank)
+        - [TechEmpower Benchmarks Power BI](https://msit.powerbi.com/view?r=eyJrIjoiYTZjMTk3YjEtMzQ3Yi00NTI5LTg5ZDItNmUyMGRlOTkwMGRlIiwidCI6IjcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NyIsImMiOjV9)
+- Runtime
+    - Can be run on CI: **Yes** CI runs require you to submit a PR against dotnet/performance
+    - Can be run locally with a single script: **No**
+    - The runtime team has a set of benchmarking guides that detail how to run the tests in dotnet/performance against local changes.
+        - [Benchmarking](https://github.com/dotnet/performance/blob/main/docs/benchmarking-workflow-dotnet-runtime.md)
+        - [Profiling](https://github.com/dotnet/performance/blob/main/docs/profiling-workflow-dotnet-runtime.md)
+
+## Proposed Workflow
+
+### Tests
+We will have two types of tests:
+
+#### Micro-Benchmarks
+A set of micro-benchmarks (written in BenchmarkDotnet) testing how much time analyzers spend computing result. Each new analyzer that ships in the SDK is expected to have a micro-benchmark that tests
+- code files that cause the analyzer to execute but not issue a a diagnostic.
+- code files that cause the analyzer to issue a diagnostic.
```

Comment:
📝 This is _not_ considered a fast path at scale.

Reply:
I don't think we want to test these at scale in micro-benchmarks. My intent is that tests follow  [these guidelines](https://github.com/dotnet/performance/blob/main/docs/microbenchmark-design-guidelines.md)

Reply:
While this is the 0.01% case we still want to know what its impact is even if it is not as common